code/modules/mob/living/simple_animal/abnormality/he/golden_false_apple.dm

```diff
@@ -59,68 +58,112 @@
+	button_icon_state = "golden_sheen_noBG"
+	check_flags = AB_CHECK_CONSCIOUS
+	transparent_when_unavailable = TRUE
+	cooldown_time = 15 SECONDS
+
+/datum/action/innate/abnormality_attack/false_attack
+	name = "Attack"
+	icon_icon = 'icons/effects/effects.dmi'
+	button_icon_state = "smash1"
+	chosen_message = "<span class='colossus'>You will now bite your target.</span>"//TODO: give it lifesteal
+	chosen_attack_num = 1
+
+/datum/action/innate/abnormality_attack/maggot_spread
+	name = "Slam"
+	icon_icon = 'ModularTegustation/Teguicons/status_sprites.dmi'
+	button_icon_state = "maggots_noBG"
+	chosen_message = "<span class='colossus'>You will now spread maggots within a wide vicinity.</span>"
+	chosen_attack_num = 2
+
+/datum/action/innate/abnormality_attack/maggot_spread2
+	name = "Lunge"
+	icon_icon = 'ModularTegustation/Teguicons/status_sprites.dmi'
+	button_icon_state = "maggots_noBG"
+	chosen_message = "<span class='colossus'>You will now spread maggots within a narrow vicinity.</span>"
+	chosen_attack_num = 3
+
+/datum/action/cooldown/gapple_pulse/Trigger()
+	if(!..())
+		return FALSE
+	if(!istype(owner, /mob/living/simple_animal/hostile/abnormality/golden_apple))
+		return FALSE
+	var/mob/living/simple_animal/hostile/abnormality/golden_apple/apple = owner
+	if(apple.IsContained()) // No more using cooldowns while contained
+		return FALSE
+	if(apple.is_maggot == TRUE)
+		to_chat(owner, span_warning("You cannot activate this ability after metamorphosis."))
+		return FALSE
+	if(apple.pulse_count == 0)
+		to_chat(owner, span_warning("You cannot activate this due to a lack of charges. Attack a hostile target to gain more charges."))
+		return FALSE
+	apple.pulse_count -= 1
+	StartCooldown()
+	apple.HealPulse(TRUE)
+	to_chat(owner, span_warning("You have [apple.pulse_count] charges remaining."))
+	return TRUE
 
 //***Simple Mob Procs***
-
-/obj/item/food/grown/apple/gold/abnormality
-	food_reagents = list(/datum/reagent/abnormality/ambrosia = 10)
-	desc = "There's something moving underneath the skin, but it still looks delicious."
-
-/datum/reagent/abnormality/ambrosia
-	name = "Ambrosia"
-	description = "A powerful serum extracted from an abnormality."
-	color = "#03FCD3"
-	taste_description = "apple juice"
-	glass_name = "glass of ambrosia"
-	glass_desc = "A glass of apple juice."
-	metabolization_rate = 3 * REAGENTS_METABOLISM//metabolizes at 24u/minute
-
-/datum/reagent/abnormality/ambrosia/on_mob_add(mob/living/L)
+/mob/living/simple_animal/hostile/abnormality/golden_apple/Initialize()
+	if(CheckCombat())//Is it R corp assault? Hit 'em with the nerf bat!
+		pulse_cooldown_time = 130 SECONDS//The duraction of the buff is 60 seconds; you can't build stacks at this rate.
+	false_attack = new /datum/action/innate/abnormality_attack/false_attack
+	maggot_attack = new /datum/action/innate/abnormality_attack/maggot_spread
+	maggot_attack2 = new /datum/action/innate/abnormality_attack/maggot_spread2
```

Comment:
unsure if these add to the abnormality correctly.

Reply:
Those are only initialized there so that they can be added by the proc HandleAbilityButtons()

Reply:
These are granted on the proc HandleAbilityButtons(), line 246